rfcs/inline_symbols.md

```diff
@@ -0,0 +1,624 @@
+value's type descriptor byte before deciding whether the value can be skipped. Additionally, the
+wrapped values themselves already specify their encoded size. This means that two of the four
+Annotation header bytes in the above example are of dubious value.
+
+This RFC adds two new encodings for annotations:
+1. An encoding that is optimized for the most common case, in which there is a single annotation.
+2. An encoding that is optimized for the case in which a value has multiple annotations.
+
+Both encodings support inline symbol definitions using the `VarInt` encoding scheme described in
+the section [*Inline symbol structs*](#0xf4-inline-symbol-structs).
+
+Because Ion 1.0's wrapper encoding cannot be shorter than 3 bytes, type descriptor bytes `0xE1` and
+`0xE2` were not legal type descriptor bytes. We use them in Ion 1.1 to represent our new encodings.
+
+##### `0xE1`: Singleton inline symbol annotation
+
+```
+            7       4 3       0
+            +---------+---------+
+Value with  |   14    |    1    |
+singleton   +---------+---------+=======================================+
+annotation  :     annotation [VarInt + optional UTF8 representation]    :
+            +-----------------------------------------------------------+
+            |                           value                           |
+            +-----------------------------------------------------------+
+```
+
+Unlike most encodings, singleton annotation encodings do not include an `L` or `Length` field. When
+a `0xE1` type descriptor byte is encountered, it is always followed by a `VarInt`-based encoding of
+the necessary annotation symbol, either as an ID or as inline UTF-8 text. (See the section [*Inline symbol
+structs*](#0xf4-inline-symbol-structs) for more detail.)
+
+Returning to our earlier example, this value:
+
+```js
+Author::"Ernest Hemingway"
+```
+
+would be encoded either using inline text:
+
+```
+       A  u  t  h  o  r        E  r  n  e  s  t     H  e  m  i  n  g  w  a  y
+E1 C6 41 75 74 68 6f 72 8E 90 45 72 6E 65 73 74 20 48 65 6D 69 6E 67 77 61 79 
+ |  |                    |  +--- VarUInt Length: 16 bytes
+ |  |                    +------ String w/VarUInt Length
+ |  +--------------------------- VarInt -6: 6 bytes of inline text
+ +------------------------------ Singleton annotation	
+```
+
+or using a symbol ID:
+
+```
+             E  r  n  e  s  t     H  e  m  i  n  g  w  a  y
+E1 CA 8E 90 45 72 6E 65 73 74 20 48 65 6D 69 6E 67 77 61 79 
+ |  |  |  +---- VarUInt Length: 16 bytes
+ |  |  +------- String w/VarUInt Length
+ |  +---- VarInt 10: symbol ID $10
+ +------- Singleton annotation
+```
+
+To skip a singleton annotation, the reader must read the `VarInt`. If it is negative, the reader
+must skip that number of bytes to move beyond the UTF8 bytes that follow it. The reader will then
+be positioned over the annotated value, which it can read or skip as needed.
+
+##### `0xE2`: Multiple inline symbol annotations
```

Comment:
Maybe call this "Optionally inline annotations" or some such to line up with the modality of the encoding.